### PR TITLE
ncbitoolkit: Create new package.

### PR DESCRIPTION
**Pull request by lingnanyuan**
Add new package: NcbiToolkit.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ncbi-toolkit/package.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    def configure_args(self):
+        return['--without-sybase', '--without-fastcgi']
+
+    def build(self, spec, prefix):
+        with working_dir(join_path('src', 'util', 'image')):
+            filter_file(r'jpeg_start_compress\(\&cinfo, true\)',
+                        'jpeg_start_compress(&cinfo, TRUE)',
+                        'image_io_jpeg.cpp')
+
+        CompilerVersion = re.sub(r'\.', "", self.compiler.version.string)
```
[the remark this answers is not in the document]
Comment (lingnanyuan) [reply var/spack/repos/builtin/packages/ncbi-toolkit/package.py]: It doesn't work, report error like this:

Process Process-1:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib64/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/home/las_lynn94/spack/lib/spack/spack/build_environment.py", line 702, in child_process
    msg = "%s: %s" % (exc_type.__name__, str(exc))
TypeError: __str__ returned non-string (type Version)

[the remark this answers is not in the document]
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/ncbi-toolkit/package.py]: To be clear, I'm suggesting to replace the entire line with this, no need for `re.sub`.

[the remark this answers is not in the document]
Comment (lingnanyuan) [reply var/spack/repos/builtin/packages/ncbi-toolkit/package.py]: Yes, I did so: CompilerVersion = self.compiler.version.joined

[the remark this answers is not in the document]
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/ncbi-toolkit/package.py]: Hmm, now I'm confused. I tried adding the following to an example package:
```python
print(self.compiler.version)                                            
print(type(self.compiler.version))                                      
print(self.compiler.version.joined)                                     
print(type(self.compiler.version.joined))                               
x = 'GCC{0}-DebugMT64'.format(self.compiler.version.joined)             
print(x)      
```
and it printed:
```
9.0.0-apple
<class 'spack.version.Version'>
900apple
<class 'spack.version.Version'>
GCC900apple-DebugMT64
```
Are you using the latest develop branch?

[the remark this answers is not in the document]
Comment (lingnanyuan) [reply var/spack/repos/builtin/packages/ncbi-toolkit/package.py]: Thanks, it works now.

**Pull request merged**